Cargo.toml

```diff
@@ -167,6 +167,7 @@ string_wizard       = { path = "./crates/string_wizard", features = ["serde"] }
 sugar_path          = { version = "1.2.0", features = ["cached_current_dir"] }
 testing_macros      = "1.0.0"
 tokio               = { version = "1.38.0", default-features = false }
+tokio_with_wasm     = { version = "0.8.2", features = ["rt"] }
```

Comment:
why need this

Reply:
https://github.com/cunarist/tokio-with-wasm/blob/180d56afc4f1e030b6116ba409de5c0c71ad0e73/tokio_with_wasm/src/glue/task/pool.rs#L75C4-L124C6

It uses webwork to support multiple spawn, and it turns out that one of the reasons wasi doesn't work is because it gets blocked, as you can see in this demo project that I'm recreating
https://github.com/rolldown/rolldown/issues/898#issuecomment-2708132359

Reply:
The wasi is blocked due to we are not using `model=reactor`, see https://github.com/rust-lang/rust/issues/137510#issuecomment-2683759021 for the context.
I fixed this in https://github.com/napi-rs/napi-rs/pull/2480, upgrade napi-build can apply this fix.

Use `tokio_with_wasm` is meaningless here because:
1. we are target to `wasm32-waspi1-threads` which has threads in our wasi polyfill: https://toyobayashi.github.io/emnapi-docs/guide/multithreaded-async.html#initialization
2. `tokio_with_wasm` requires `nightly` toolchain and needs pass `-Z build-std=std,panic_abort` and `RUSTFLAGS="-C target-feature=+atomics,+bulk-memory,+mutable-globals"` to rust but we are using stable Rust.

Reply:
Experts are experts, awesome!

Reply:
![image](https://github.com/user-attachments/assets/8c58d70a-6832-4379-908c-e65e282ef3d2)
![image](https://github.com/user-attachments/assets/4b85c3c5-aeb0-4b52-8a59-062c5faf48e6)
![image](https://github.com/user-attachments/assets/a5e63bd0-627a-4089-b0cf-8e7bf0c0deb9)

Is there something wrong with my configuration? Why is it still blocked?

https://github.com/suxin2017/napi-bind

Reply:
You can't call `tokio::spawn` on the Node.js main thread, it will call [`memory.atomic.wait32 instruction`](https://webassembly.github.io/threads/core/syntax/instructions.html#syntax-instr-atomic-memory) under the hood and block the main thread forever. You can use `std::thread::spawn` + `tokio::block_on` as a workaround like this: https://github.com/napi-rs/napi-rs/blob/main/crates/napi/src/tokio_runtime.rs#L216-L221

Note, this limitation can be resolved by using `MultiThread` tokio `Runtime` on wasm32, I'm working on this in https://github.com/napi-rs/napi-rs/pull/2501